server/raft.go

```diff
@@ -369,6 +369,31 @@ func (s *Server) lookupRaftNode(group string) RaftNode {
 	return n
 }
 
+func (s *Server) transferRaftLeaders() bool {
+	if s == nil {
+		return false
+	}
+
+	var nodes []RaftNode
+	s.rnMu.RLock()
+	if len(s.raftNodes) > 0 {
+		s.Noticef("Transferring any raft leaders")
+	}
+	for _, n := range s.raftNodes {
+		nodes = append(nodes, n)
+	}
+	s.rnMu.RUnlock()
```

Comment:
But is it not a problem that we are releasing here? Meaning say we capture 3 nodes, but then 1 is added after that, and it happens to be the leader?

Should we also collect only leaders here? or we can't because node.Leader() cannot run under s.rnMu lock?

Reply:
Correct..